src/gsy_e/models/strategy/__init__.py

```diff
@@ -374,13 +374,13 @@ def can_offer_be_posted(
 
         if replace_existing:
             # Do not consider previous open offers, since they would be replaced by the current one
-            posted_offer_energy = self.sold_offer_energy(market.id, time_slot)
```

Comment:
I really wonder why did this not show up till now. I assume because this was only called from the external strategies and the template strategies that we usually test our code against do not use it right ?

Reply:
I asked myself the same and concluded equally. This may even fix some issues the research team still have with the external strategies.

Reply:
I'm not sure I understood the fix. In my head, if the new offer has `replace_existing`, it should replace only the open offers that have not been traded yet, not all offers. So the energy should be equal to the energy of the offers that have been traded so far, not `0`. Could you please explain this part?

Reply:
Yes, you are right, and this was/is already take care of in line 383.
When setting `replace_existing=True` we only ignore the already posted offers, as they will be replaces in the following. I hope this explains it. Happy to be more verbose if you like.

Reply:
On line 383 we added the `time_slot` parameter to `sold_offer_energy()`. But this shouldn't change the behaviour as far as I know (it only affects future markets). I think I need the verbose explanation :)

Reply:
Ha, I just had another (appr. 1000th) look at this and indeed, in line 383 we subtract the `sold_offer_energy` from the posted_energy which made the `replace_existing=True` case work, because we first add and then subtract the same number. But for the `replace_existing=False` case. we were not taking into account the `sold_offer_energy` .
So, this will be my proposal to fix both cases:
```python
    if replace_existing:
            posted_offer_energy = 0.0
        else:
            posted_offer_energy = self.posted_offer_energy(market.id, time_slot)

        posted_energy = (offer_energy
                         + posted_offer_energy
                         + self.sold_offer_energy(market.id, time_slot))
```
What do you think?

Reply:
So if `replace_existing is True`, then our posted energy is sum of the energy of the 
- current offer
- the previous offers we've already traded.

If `replace_existing is False`, then our posted energy is the sum of the energy of the 
- current offer
- the previous offers we haven't traded yet
- the previous offers we have already traded.

Is this correct? If this is right, the snippet you posted seems good to me 👍

Reply:
Yes, that is correct. There are tests that I will extend to catch all corner cases. Thanks!